main/lsp/requests/completion.cc

```diff
@@ -912,9 +906,18 @@ vector<unique_ptr<CompletionItem>> CompletionTask::getCompletionItems(LSPTypeche
             getCompletionItemForLocal(gs, this->config, similarLocal, params.queryLoc, params.prefix, items.size()));
     }
     for (auto &similarMethod : dedupedSimilarMethods) {
+        // Even though we might have one or more TypeConstraints on the DispatchResult that triggered this completion
+        // request, those constraints are the result of solving the current method. These new methods we're about to
+        // suggest are their own methods with their own type variables, so it doesn't make sense to use the old
+        // constraint for the new methods.
+        //
+        // What this means in practice is that the prettified `sig` in the completion documentation will show
+        // `T.type_parameter(:U)` instead of a solved type.
+        auto constr = nullptr;
```

Comment:
Thanks for this comment, it really cleared up the deletions above 👍 

Sort of amazing that `nullptr` works here, is the deduced type `std::nullptr_t`?

Reply:
> Sort of amazing that nullptr works here, is the deduced type std::nullptr_t?

yep